src/panels/shopping-list/ha-panel-shopping-list.ts

```diff
@@ -0,0 +1,107 @@
+import "@polymer/app-layout/app-header/app-header";
+import "@polymer/app-layout/app-toolbar/app-toolbar";
+import {
+  css,
+  CSSResultArray,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+  PropertyValues,
+  TemplateResult,
+} from "lit-element";
+
+import { HomeAssistant } from "../../types";
+import { haStyle } from "../../resources/styles";
+import { createCardElement } from "../lovelace/create-element/create-card-element";
+import { LovelaceCard } from "../lovelace/types";
+import { HuiErrorCard } from "../lovelace/cards/hui-error-card";
+
+import "../../components/ha-menu-button";
+import "../../layouts/ha-app-layout";
+
+@customElement("ha-panel-shopping-list")
+class PanelShoppingList extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean, reflect: true }) public narrow!: boolean;
+
+  @internalProperty() private _card!: LovelaceCard | HuiErrorCard;
+
+  protected firstUpdated(changedProperties: PropertyValues): void {
+    super.firstUpdated(changedProperties);
+
+    this._card = createCardElement({ type: "shopping-list" }) as LovelaceCard;
+    this._card.hass = this.hass;
+  }
+
+  protected updated(changedProperties: PropertyValues): void {
+    super.updated(changedProperties);
+
+    if (changedProperties.has("hass")) {
+      this._card.hass = this.hass;
+    }
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <ha-app-layout>
+        <app-header fixed slot="header">
+          <app-toolbar>
+            <ha-menu-button
+              .hass=${this.hass}
+              .narrow=${this.narrow}
+            ></ha-menu-button>
+            <div main-title>${this.hass.localize("panel.shopping_list")}</div>
+            </div>
+          </app-toolbar>
+        </app-header>
+        <div id="columns">
+          <div class="column">
+            ${this._card}
+          </div>
+        </div>
+      </ha-app-layout>
+    `;
+  }
+
+  static get styles(): CSSResultArray {
+    return [
+      haStyle,
+      css`
+        :host {
+          --mdc-theme-primary: var(--app-header-text-color);
+          display: block;
+          height: 100%;
+        }
+        :host([narrow]) app-toolbar mwc-button {
+          width: 65px;
+        }
+        .heading {
+          overflow: hidden;
+          white-space: nowrap;
+          margin-top: 4px;
+        }
+        #columns {
+          display: flex;
+          flex-direction: row;
+          justify-content: center;
+          margin-left: 4px;
+          margin-right: 4px;
+        }
+        .column {
+          flex: 1 0 0;
+          max-width: 500px;
```

Comment:
Should we add a min-width and a columns/column? Why not just will width?

![image](https://user-images.githubusercontent.com/5662298/97815400-033a0580-1c8e-11eb-9278-ffe7525b0a8c.png)

Reply:
I had that initially and it just seemed way too big. Old panel had it centered as I have it now and thought it looked a bit better. 🤷‍♂️I'm really indifferent though.

Reply:
I agree that full width is not what we want. It means the buttons at the end are too far away from the actual content.